packages/tinacms/src/toolkit/fields/components/reference/reference-select.tsx

```diff
@@ -182,33 +183,32 @@ const ComboboxDemo: React.FC<ReferenceSelectProps> = ({
                     heading={collection}
                   >
                     <CommandList>
-                      {edges.map(
-                        ({
-                          node: {
-                            id,
-                            _internalSys: { filename },
-                            _values: { title, name },
-                          },
-                        }) => (
+                      {edges.map(({ node }) => {
+                        const { id, _values } = node
+
+                        return (
                           <CommandItem
                             key={`${id}-option`}
-                            value={id || title}
+                            value={id}
```

Comment:
Can we make this better by adding another `search` prop on the reference type that the user can input for what to search on.Either a string or this could even be a function that the user controls completely

Reply:
In addition to making search customizable, Ben and I discussed what it would look like to improve search behavior. We'll want to make this paginated in the future so we'll need to implement search from the GraphQL query. To do that we'll need to use the `filter` argument on the the documents

```graphql
query ($collection: String!, $filter: DocumentFilter){
  collection(collection: $collection) {
    documents(first: -1, filter: $filter) {
```
Where `DocumentFilter` supports querying on the colletion's fields. We can make use of the `isTitle` field to give them title search behavior by default.

Reply:
As per conversation with @jeffsee55 , we will tackle the custom search in a different pr (current pr is for custom reference e component and the search still work based on filename )